File: compressor/analysis-tools/filenameparams.py
# Functions to get params from filenames
import os.path as path


class FnameParams(object):
    def __init__(self, fname):
        fn = path.basename(fname)
        parts = fname.split("_")
        if parts[0] != "output":
            raise ValueError("Not an output file.")
        self.natoms = int(parts[1])
        self.ext_frac = float(parts[2])
        self.njit = int(parts[3])
        self.sorted = parts[4] == "sorted"
        self.aatype = int(parts[5][0])
        self.eedist = (self.natoms - 1) * self.ext_frac

File: compressor/analysis-jl/filenameparams.jl
## A class to extract simulation parameter information from a filename.
#  Note that this is a skeleton function and will likely be edited to
#  fit the file that's being worked on.

struct FNameParam
    natoms::Int
    ext_frac::Float64
    njit::Int
    sorted::Bool
    aatype::Int
    eedist::Float64
end

FNameParam(filename) = begin
    fn = basename(filename)
    parts = split(fn, '_')
    natoms = parse(parts[2], Int)
    ext_frac = parse(parts[3], Float64)
    njit = parse(parts[4], Int)
    sorted = parts[5] == "sorted"
    aatype = parse(parts[6][1], Int)
    eedist = (natoms - 1) * ext_frac
    FNameParam(natoms, ext_frac, njit, sorted, aatype, eedist)
end